chapters/en/slides/module1/module1_02.Rmd

```diff
@@ -0,0 +1,280 @@
+in terms of *data*, *graphical marks*, and *encoding channels*,
+rather than having to specify *how* to implement the visualization in terms of for-loops, low-level drawing commands, *etc*.
+For example,
+you would say "color my data by the column 'country'"
+instead of "go through this data frame and plot any observations of country1 in blue, any observations of country in red, etc".
+
+Declarative visualization tools lets you think about **data and relationship**,
+rather than **plot construction details**.
+A key idea is that you declare links between data fields and visual encoding channels,
+such as the x-axis, y-axis, color, *etc*.
+The rest of the plot details are handled automatically.
+
+---
+
+## The Python plotting landscape
+
+<img src="/module1/py-plotting-landscape.png" alt="The Python plotting landscape" width="100%"></img>
+
+Notes:
+In this image you can see the most commonly used Python plotting packages.
+There are many more,
+but these are the ones you are the most likely to hear about,
+so it is good to know that they exist.
+As you can see there are several high and low level language,
+so how to we chose?
+In this course we have chosen to use Altair,
+because it is a powerful high level visualization tool 
+with a clear and consistent syntax
+that also allows us to add interactive components to our plots,
+such as tooltips and selections.
+
+---
+
+## A high level grammar of graphics helps us compose plots effectively
+
+- Simple grammatical components combine to create visualizations.
+- The Altair visualization grammar consist of three main components:
+    1. Create a chart.
+    2. Add a graphical mark.
+    3. Encode dataframe columns as visual channels.
+- In code, it looks like this: `Chart(data).mark().encode(x, y, ...)`.
+
+Notes:
+The declarative plotting concept can be implemented in different ways.
+Here we will describe how a visualization grammar
+can be used similarly to how grammar in regular language
+is used to build complex linguistic constructs.
+A wide range of simple to sophisticated visualizations can be created using a concise grammar.
+Thanks to this functional way of interfacing with data,
+only minimal changes are required if the underlying data change or to change the type of plot.
+
+One of the most prominent declarative statistical visualization libraries is [Altair](https://altair-viz.github.io/).
+Altair offer a powerful and concise visualization grammar for quickly building a wide range of statistical graphics.
+In brief, you first create a chart,
+then you indicate which graphical marks should represent the data (points, lines, etc)
+and finally you encode your dataframe colums as different visual channels (x, y, color, etc).
+
+Enough talking,
+let's code!
+
+---
+
+## Sample data can be found in Altair's companion package vega_datasets
+
+```{python}
+from vega_datasets import data
+
+cars = data.cars()
```

Comment:
I don't mean to be negative, however, could we pick a more interesting data set? We need to get folks excited and motivated to stick around! Maybe a pre-amble also of a question we'd like to answer, and then we introduce the altair code slowly as you do now to answer it?

In online courses it is important to motivate, or we lose learners. There's not the same level of engagement. You could borrow a question and viz from Introduction to Data Science? See here: https://ubc-dsci.github.io/introduction-to-datascience/viz.html#choosing-the-visualization

Reply:
I kept the car data because of time concerns and that I want to show `vega_datasets` but I have written more to build it up similar to what you do in the link above including a questions we might have of the data so that it hopefully feels more interesting.